### PR TITLE
Load Secret Variables

### PR DESCRIPTION
**Pull request by MattBodily**
Added in the ability to load values into variables marked as secret.

### CODE REVIEW

**Comment by sandorfr** [comment]
That seems interesting however, if your files are in the source control they are hardly a secret?

**Comment by sandorfr** [comment]
would you mind rebasing on the last commit, I think your build issues were related to the obsolete build scripts and typings.

**Comment by MattBodily** [comment]
There are cases where a build or release may need to load files from other
locations, handle the results of a REST call, etc. For instance, the
current REST plugin on the marketplace will write results to a file which
could then be loaded to variables and the file deleted.

**Comment by MattBodily** [comment]
Yeah, I'll let the rebase done.

**Comment by MattBodily** [comment]
Rebase done and yep, that fixed things. Sorry about that.

File: Tasks/LoadJsonVariables/task.json [hunk 1]
```diff
@@ -36,6 +36,14 @@
             "defaultValue": "",
             "required": true,
             "helpMarkDown": "Loaded variables will be named PREFIX_VariableName"
+        }, 
+        {
+            "name" : "MarkSecret",
```
Comment (sandorfr) [inline Tasks/LoadJsonVariables/task.json]: This needs to be added to other task.json files too

File: Tasks/LoadYamlVariables/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "version": "0.1.0",
   "dependencies": {
     "fs-extra": "0.30.0",
-    "vsts-task-lib": "0.9.6",
+    "vsts-task-lib": "2.4.0",
     "js-yaml": "3.6.1"
   }
-}
+}
```
Comment (sandorfr) [inline Tasks/LoadYamlVariables/package.json]: please restore the extra line

File: Tasks/LoadPlistVariables/package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "version": "0.1.0",
   "dependencies": {
     "fs-extra": "0.30.0",
-    "vsts-task-lib": "0.9.6",
+    "vsts-task-lib": "2.4.0",
     "plist": "2.0.1"
   }
-}
+}
```
Comment (sandorfr) [inline Tasks/LoadPlistVariables/package.json]: please restore the extra line

File: Tasks/LoadJsonVariables/loadjsonvariables.ts [hunk 4]
```diff
@@ -6,11 +6,12 @@ import {recursiveProcessing} from './common/expandJObject';
 try { 
     var source = tl.getPathInput("JsonSource");
     var variablePrefix = tl.getInput("VariablePrefix");
+    var isSecret = tl.getBoolInput("MarkSecret");
```
Comment (sandorfr) [inline Tasks/LoadJsonVariables/loadjsonvariables.ts]: Maybe name this `MarkAsSecret` instead

File: Tasks/LoadJsonVariables/task.json (hunk 1 above)
Comment (sandorfr) [reply Tasks/LoadJsonVariables/task.json]: And name it `MarkAsSecret`

**Comment by MattBodily** [comment]
Updates to include requested changes have been made.

**Comment by MattBodily** [comment]
No problem. Thanks for working through things and getting it merged. I assume there isn't anything I need to be involved with to get this updated in the marketplace.

**Pull request merged**